http/native_server.ts

```diff
@@ -0,0 +1,586 @@
+    });
+
+    return await this.serve(listener);
+  }
+
+  /**
+   * Immediately close the server listeners and associated HTTP connections.
+   *
+   * Throws a server closed error if called after the server has been closed.
+   *
+   * @throws {Deno.errors.Http} When the server has already been closed.
+   */
+  close(): void {
+    if (this.#closed) {
+      throw ERROR_SERVER_CLOSED;
+    }
+
+    this.#closed = true;
+
+    for (const listener of this.#listeners) {
+      try {
+        listener.close();
+      } catch (error) {
+        if (!(error instanceof Deno.errors.BadResource)) {
+          throw error;
+        }
+      }
+    }
+
+    this.#listeners.clear();
+
+    for (const httpConn of this.#httpConnections) {
+      try {
+        httpConn.close();
+      } catch (error) {
+        if (!(error instanceof Deno.errors.BadResource)) {
+          throw error;
+        }
+      }
+    }
+
+    this.#httpConnections.clear();
+  }
+
+  /**
+   * Get whether the server is closed.
+   */
+  get closed(): boolean {
+    return this.#closed;
+  }
+
+  /**
+   * Serves all HTTP requests on a single TCP connection.
+   *
+   * @param {Deno.HttpConn} httpConn The HTTP connection to yield requests from.
+   * @param {Deno.Conn} conn The TCP connection.
+   * @private
+   */
+  private async serveHttp(
+    httpConn: Deno.HttpConn,
+    conn: Deno.Conn,
+  ): Promise<void> {
+    let requestEvent: Deno.RequestEvent | null = null;
+
+    while (!this.#closed) {
+      try {
+        // Yield the new HTTP request on the connection.
+        requestEvent = await httpConn.nextRequest();
+      } catch {
+        // Connection has been closed.
+        break;
+      }
+
+      if (requestEvent === null) {
+        // Connection has been closed.
+        break;
+      }
+
+      // Wrap request event so can gracefully handle and await async ops.
+      const serverRequest = new ServerRequest(requestEvent, conn);
+
+      // Handle the request event, generating a response.
+      let response: Response;
+      try {
+        response = await this.#handler(
+          serverRequest.request,
+          serverRequest.connInfo,
+        );
+      } catch {
+        // If the handler throws then it is assumed that the impact of the
+        // error is isolated to the individual request, so we close the
+        // connection.
+        break;
+      }
+
+      // Send the response.
+      serverRequest.respondWith(response);
+
+      try {
+        // Wait for the request to be processed before we accept a new request
+        // on this connection.
+        await serverRequest.done;
+      } catch {
+        // Connection has been closed.
+        break;
+      }
+
+      requestEvent = null;
+    }
+
+    this.untrackHttpConnection(httpConn);
+
+    try {
+      httpConn.close();
+    } catch (error) {
+      if (!(error instanceof Deno.errors.BadResource)) {
+        throw error;
+      }
+    }
+
+    if (requestEvent) {
```

Comment:
TODO: Should be fixed in Deno `1.13.1`

REF: https://github.com/denoland/deno/issues/11595#issuecomment-899414934

Reply:
Wasn’t resolved, raised https://github.com/denoland/deno/issues/11743